Geth panic when reporting to ethstats while syncing

#### System information

Geth version: `1.8.0-unstable`
OS & Version: Linux
Commit hash : c786f75389121c29d4380c53a013759254457776

#### Expected behaviour

Geth sync and report no history to ethstats

#### Actual behaviour

Geth fail to report and panic

#### Steps to reproduce the behaviour

# Start with a fresh blockchain
$> rm -rf ~/.ethereum/testnet/geth/chaindata
# Start geth with ethstats option
$> ./build/bin/geth --testnet --fast --ethstats GETH:PASSWORD@127.0.0.1:3000
# Wait for ethstats to do an [history request]

Panic

#### Backtrace

````
INFO [01-12|17:12:42] Starting peer-to-peer node               instance=Geth/v1.8.0-unstable-72e70bce/linux-amd64/go1.9.2
INFO [01-12|17:12:43] Loaded most recent local header          number=45797 hash=7cac75…272854 td=4187812833084
INFO [01-12|17:12:43] Loaded most recent local full block      number=0     hash=419410…ca4a2d td=1048576
INFO [01-12|17:12:43] Loaded most recent local fast block      number=45018 hash=2b3801…76d6e2 td=4112261950300
INFO [01-12|17:12:43] Loaded local transaction journal         transactions=0 dropped=0
INFO [01-12|17:12:43] Upgrading chain index                    type=bloombits percentage=81
INFO [01-12|17:13:35] Block synchronisation started 
INFO [01-12|17:13:52] Imported new block headers               count=0 elapsed=15.515ms number=45210 hash=282df7…4ce337 ignored=192
INFO [01-12|17:13:52] Imported new block headers               count=0 elapsed=16.696ms number=45402 hash=f4f7c9…6b7627 ignored=192
INFO [01-12|17:13:53] Imported new block receipts              count=2 elapsed=1.115ms  bytes=6031 number=45020 hash=28f858…be5628 ignored=0
[...]
INFO [01-12|17:13:59] Imported new block receipts              count=205 elapsed=37.285ms bytes=305126 number=46330 hash=ad4c74…89c92a ignored=0
INFO [01-12|17:13:59] Imported new block headers               count=192 elapsed=83.064ms number=47130 hash=17bd27…82c9b2 ignored=0
WARN [01-12|17:14:00] Invalid stats history request            msg=false
panic: runtime error: slice bounds out of range

goroutine 145 [running]:
github.com/ethereum/go-ethereum/ethstats.(*Service).reportHistory(0xc420310b60, 0xc4203425a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethstats/ethstats.go:615 +0x8a0
github.com/ethereum/go-ethereum/ethstats.(*Service).loop(0xc420310b60)
	/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethstats/ethstats.go:255 +0xc89
created by github.com/ethereum/go-ethereum/ethstats.(*Service).Start
	/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethstats/ethstats.go:123 +0x57
````

--- a/ethstats/ethstats.go
+++ b/ethstats/ethstats.go
@@ -613,6 +613,7 @@ func (s *Service) reportHistory(conn *websocket.Conn, list []uint64) error {
 		}
 		// Ran out of blocks, cut the report short and send
 		history = history[len(history)-i:]
+		break
 	}
 	// Assemble the history report and send it to the server
 	if len(history) > 0 {

